group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntime.java

```diff
@@ -543,8 +543,14 @@ private void transitionTo(
 
                 case ACTIVE:
                     state = CoordinatorState.ACTIVE;
-                    snapshotRegistry.getOrCreateSnapshot(0);
                     partitionWriter.registerListener(tp, highWatermarklistener);
+
+                    // If the partition did not contain any records, we would not have generated a snapshot
+                    // while loading.
+                    if (lastWrittenOffset == -1L) {
+                        updateLastWrittenOffset(0);
+                        updateLastCommittedOffset(0);
+                    }
```

Comment:
I was wondering whether we could just keep 0 as the default values for both (In LOADING state) and create the base snapshot there. Then, we don't have to handle a special case here.

Reply:
for both offsets, are you saying that we won't hit the case where we update HWM or last written offset to 0 while loading? SInce we throw an exception in both methods
```
            if (offset <= lastWrittenOffset) {
                throw new IllegalStateException("New last written offset " + offset + " of " + tp +
                    " must be larger than " + lastWrittenOffset + ".");
            }
```
Which is why I initialized them as -1L

Reply:
i reverted this and instead created a snapshot at epoch 0 when transitioning to LOADING state. it doesn't make sense to create a snapshot at epoch 0 after loading.

i also changed the exception conditions to be `>` and not `>=` since we do expect lastCommittedOffset and lastWrittenOffset to sometimes be updated to the same value.